source/SemVerAnalyzer/Nuget/NugetClient.cs

```diff
@@ -21,7 +25,32 @@ public async Task<byte[]> GetAssemblyBytesFromPackage(string packageName, string
 			try
 			{
 				using var client = new HttpClient();
-				using var request = new HttpRequestMessage(HttpMethod.Get, Path.Join(_config.RepositoryUrl, packageName));
+				
+				using var feedRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(_config.RepositoryUrl, "v3", "index.json"));
+				using var feedResponse = await client.SendAsync(feedRequest);
+				if (!feedResponse.IsSuccessStatusCode)
+				{
+					var content = await feedResponse.Content.ReadAsStringAsync();
+					comments.Add($"Error retrieving Nuget feed:\n{content}");
+					return null;
+				}
+
+				var feed = JsonSerializer.Deserialize<NugetPackageFeed>(await feedResponse.Content.ReadAsStringAsync());
+				var packageBaseAddress = feed.Resources.Single(r => r.Type.StartsWith("PackageBaseAddress")).Id;
+				
+				using var versionRequest = new HttpRequestMessage(HttpMethod.Get, Path.Combine(packageBaseAddress, packageName, "index.json"));
+				using var versionResponse = await client.SendAsync(versionRequest);
+				if (!versionResponse.IsSuccessStatusCode)
+				{
+					var content = await versionResponse.Content.ReadAsStringAsync();
+					comments.Add($"Error retrieving package versions:\n{content}");
+					return null;
+				}
+
+				var versions = JsonSerializer.Deserialize<VersionsFeed>(await versionResponse.Content.ReadAsStringAsync()).Versions;
+				var highestVersion = versions.OrderByDescending(v => v.MajorVersion()).ThenByDescending(v => v.MinorVersion()).ThenByDescending(v => v.PatchVersion()).First();
```

Comment:
Each one of these extension methods parses the version string.  I think it might be better to parse it once into a class and use the properties given to you.  In addition, some packages may have pre-release suffixes after the patch version.

Reply:
Good thinking, and good to know - I'll look into this

Reply:
If the struct (or if you have to, the class) they get parsed into is `IComparable<>` then you can then just `.OrderByDescending(str => str.ToVersion())`.